client/blocks/comments/post-comment-list.jsx

```diff
@@ -56,25 +69,65 @@ class PostCommentList extends React.Component {
 	};
 
 	componentWillMount() {
-		const { post: { ID: postId, site_ID: siteId } } = this.props;
+		const {
+			post: { ID: postId, site_ID: siteId },
+			commentsFilter: status,
+			startingCommentId,
+			commentsFetchingStatus: { haveReceivedBeforeAndAfter },
+		} = this.props;
 
-		this.props.requestPostComments( siteId, postId, this.props.commentsFilter );
+		if (
+			startingCommentId &&
+			this.props.commentsTree[ startingCommentId ] &&
+			! haveReceivedBeforeAndAfter &&
+			! this.alreadyLoadedInitialSet
+		) {
+			this.viewEarlierCommentsHandler();
+			this.viewLaterCommentsHandler();
+			this.alreadyLoadedInitialSet = true;
+		} else if ( startingCommentId ) {
+			this.props.requestComment( { siteId, commentId: this.props.startingCommentId } );
+		} else {
+			this.props.requestPostComments( { siteId, postId, status } );
+		}
 	}
 
 	componentWillReceiveProps( nextProps ) {
 		const nextSiteId = get( nextProps, 'post.site_ID' );
 		const nextPostId = get( nextProps, 'post.ID' );
 		const nextCommentsFilter = get( nextProps, 'commentsFilter' );
+		const nextStartingCommentId = get( nextProps, 'startingCommentId' );
+		const { hasReceivedBefore, hasReceivedAfter } = get( nextProps, 'commentsFetchingStatus' );
 
 		if (
```

Comment:
its messy as hell, but this creates the 2-step comment loading behavior (first comment + page before/after).  I can probably clear it up a bit w/ good function names